lib/realtime/block_waiver.ex

```diff
@@ -94,22 +101,25 @@ defmodule Realtime.BlockWaiver do
   defp bank_working([], working), do: [working]
   defp bank_working(acc, working), do: acc ++ [working]
 
-  @spec same_remark?(map(), map()) :: boolean
-  defp same_remark?(%StopTimeUpdate{remark: remark1}, %StopTimeUpdate{remark: remark2}),
-    do: remark1 == remark2
+  @spec same_cause?(StopTimeUpdate.t(), StopTimeUpdate.t()) :: boolean
+  defp same_cause?(stu1, stu2) do
+    stu1.cause_id == stu2.cause_id and stu1.remark == stu2.remark
```

Comment:
@skyqrose is `remark` a free-text field here, since if so checking for equality can be problematic

Reply:
Yes, it is free text. How can it be problematic?

Reply:
People, even if they're trying to be, are not 100% consistent about how they enter data, with regards to capitalization, whitespace, punctuation, etc. So you have issues where things that "should" be identical like `"Foo Bar"` and `"Foo bar"` are not so.

But actually, I think I heard something about this remark value ultimately originating in a dropdown so perhaps it's OK after all?

Reply:
cause_id/_description are from the dropdown. `remark` is from the accompanying free text.

It's entered once, and then duplicated by TransitMaster software across all the stops, so I don't think small human errors would really be a problem with this equality check. If it was entered by a human twice with a small difference (maybe for two halves of the run), I'd be okay with it being grouped into two separate block waivers.